### PR TITLE
feat: http raw support for Go

### PR DESCRIPTION
**Pull request by matejvasek**

### CODE REVIEW

**Comment by matejvasek** [comment]
@lkingland  This will add support for two types of signature
```go
func (res http.ResponseWriter, req *http.Request)
func (ctx context.Context, res http.ResponseWriter, req *http.Request)
```
Note they do not return any value.

**Comment by matejvasek** [comment]
related to https://github.com/boson-project/faas/issues/93

File: buildpacks/go/faas/main.go [hunk 1]
```diff
@@ -33,23 +43,92 @@ func main() {
...
+	if fnType.Kind() != reflect.Func {
+		return nil, errors.New("must pass a function to handle events/http")
+	}
+
+	if fnType.NumOut() > 0 {
+		return nil, nil
+	}
+
+	if fnType.In(0) == httpResponseWriterType && fnType.In(1) == httpRequestPtrType {
+		if f, ok := handler.(func(rw http.ResponseWriter, req *http.Request)); ok {
```
Comment (matejvasek) [inline buildpacks/go/faas/main.go:116]: This type assert is probably sufficient and the reflective check is not necessary.

File: buildpacks/go/faas/main.go [hunk 2]
```diff
@@ -33,23 +43,92 @@ func main() {
...
+			return http.HandlerFunc(f), nil
+		}
+	}
+	if fnType.In(0) == contextType &&
+		fnType.In(1) == httpResponseWriterType &&
+		fnType.In(2) == httpRequestPtrType {
+		if f, ok := handler.(func(ctx context.Context, rw http.ResponseWriter, req *http.Request)); ok {
```
Comment (matejvasek) [inline buildpacks/go/faas/main.go:123]: This type assert is probably sufficient and the reflective check is not necessary.

**Review by lance — COMMENTED** [review]
LGTM

**Review by lkingland — APPROVED** [review]
Thanks for the addition @matejvasek

**Pull request merged**